onnxruntime/python/onnxruntime_pybind_state_common.h

```diff
@@ -294,6 +299,74 @@ inline AllocatorPtr& GetAllocator() {
   return alloc;
 }
 
+// This class exposes SparseTensor to Python
+// The class serves two major purposes
+// - to be able to map numpy arrays memory and use it on input, this serves as a reference holder
+//   so incoming arrays do not disappear
+// - to be able to expose SparseTensor returned from run method
+class PySparseTensor {
+ public:
+  /// <summary>
+  /// Use this constructor when you created a SparseTensor instance which is backed
+  /// by python array storage and it important that they stay alive while this object is
+  /// alive
+  /// </summary>
+  /// <param name="instance">a fully constructed and populated instance of SparseTensor</param>
+  /// <param name="storage">a collection reference guards</param>
+  PySparseTensor(std::unique_ptr<SparseTensor>&& instance,
+                 std::vector<pybind11::object>&& storage)
+      : backing_storage_(std::move(storage)), ort_value_() {
+    Init(std::move(instance));
+  }
+
+  /// <summary>
+  /// Same as above but no backing storage as SparseTensor owns the memory
+  /// </summary>
+  /// <param name="instance"></param>
+  explicit PySparseTensor(std::unique_ptr<SparseTensor>&& instance)
+      : backing_storage_(), ort_value_() {
+    Init(std::move(instance));
+  }
+
+  explicit PySparseTensor(const OrtValue& ort_value)
+      : backing_storage_(), ort_value_(ort_value) {}
+
+  PySparseTensor(const PySparseTensor&) = delete;
+  PySparseTensor& operator=(const PySparseTensor&) = delete;
+
+  PySparseTensor(PySparseTensor&& o) noexcept {
+    *this = std::move(o);
+  }
+
+  PySparseTensor& operator=(PySparseTensor&& o) noexcept {
+    ort_value_ = std::move(o.ort_value_);
+    backing_storage_ = std::move(o.backing_storage_);
+    return *this;
+  }
+
+  ~PySparseTensor();
+
+  const SparseTensor& Instance() const {
+    return ort_value_.Get<SparseTensor>();
+  }
+
+  SparseTensor* Instance() {
+    return ort_value_.GetMutable<SparseTensor>();
+  }
+
```

Comment:
Why do we return a pointer here but a reference from the const version of Instance?

Reply:
Removed the overload as not needed